lms/templates/student_account/account_settings.html

```diff
@@ -27,12 +32,17 @@
 
 <%block name="js_extra">
 <%static:require_module module_name="js/student_account/views/account_settings_factory" class_name="AccountSettingsFactory">
-    var fieldsData = ${ json.dumps(fields) };
-    var authData = ${ json.dumps(auth) };
-    var platformName = ${json.dumps(static.get_platform_name())};
+    var fieldsData = ${ fields | n, dump_js_escaped_json };
+    var authData = ${ auth | n, dump_js_escaped_json };
+    var platformName = ${ static.get_platform_name() | n, dump_js_escaped_json };
 
     AccountSettingsFactory(
-        fieldsData, authData, '${user_accounts_api_url}', '${user_preferences_api_url}', ${user.id}, platformName
+        fieldsData, 
+        authData,
+        '${ user_accounts_api_url | n, js_escaped_string }',
+        '${ user_preferences_api_url | n, js_escaped_string }',
```

Comment:
WRT http://draft-safe-templates-2016-03-23.readthedocs.org/en/latest/conventions/safe_templates.html#determining-the-context-in-mako, should I be hunting these down in python to make sure they are getting validated properly?  If so, do we have a `validate_url()` function somewhere?  That doesn't seem like functionality I should hack on *in situ*.